rust/worker/benches/fulltext_index.rs

```diff
@@ -0,0 +1,176 @@
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+
+use anyhow::Result;
+use chroma_benchmark_datasets::{
+    datasets::{ms_marco_queries::MicrosoftMarcoQueriesDataset, scidocs::SciDocsDataset},
+    types::DocumentDataset,
+};
+use chroma_blockstore::{arrow::provider::ArrowBlockfileProvider, provider::BlockfileProvider};
+use chroma_cache::{
+    cache::Cache,
+    config::{CacheConfig, UnboundedCacheConfig},
+};
+use chroma_storage::{local::LocalStorage, Storage};
+use chroma_types::{
+    Chunk, DirectDocumentComparison, LogRecord, WhereDocument, WhereDocumentOperator,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion, Throughput};
+use futures::StreamExt;
+use uuid::Uuid;
+use worker::segment::{
+    metadata_segment::{MetadataSegmentReader, MetadataSegmentWriter},
+    types::{LogMaterializer, SegmentFlusher, SegmentWriter},
+};
+
+mod dataset_utilities;
+use dataset_utilities::{get_document_dataset, get_document_query_dataset_pair};
+
+async fn get_log_chunk<T: DocumentDataset>(corpus: &T) -> Chunk<LogRecord> {
+    let log_records = corpus
+        .create_log_stream(|_| async { vec![0.0, 0.0, 0.0] })
+        .await
+        .expect("Failed to create log stream")
+        .filter_map(|record| async {
+            match record {
+                Ok(record) => Some(record),
+                Err(_) => None,
+            }
+        })
+        .collect::<Vec<_>>()
+        .await;
+
+    Chunk::new(log_records.into())
+}
+
+async fn compact_log_to_storage(
+    blockfile_provider: &BlockfileProvider,
+    chunk: Chunk<LogRecord>,
+) -> Result<chroma_types::Segment> {
+    let materializer = LogMaterializer::new(None, chunk, None);
+
+    let mut metadata_segment = chroma_types::Segment {
+        id: Uuid::from_str("00000000-0000-0000-0000-000000000001").expect("parse error"),
+        r#type: chroma_types::SegmentType::BlockfileMetadata,
+        scope: chroma_types::SegmentScope::METADATA,
+        collection: Uuid::from_str("00000000-0000-0000-0000-000000000000").expect("parse error"),
+        metadata: None,
+        file_path: HashMap::new(),
+    };
+
+    let mut metadata_writer =
+        MetadataSegmentWriter::from_segment(&metadata_segment, &blockfile_provider)
+            .await
+            .expect("Error creating segment writer");
+
+    metadata_writer
+        .apply_materialized_log_chunk(materializer.materialize().await.unwrap())
+        .await
+        .unwrap();
+    metadata_writer.write_to_blockfiles().await.unwrap();
+    let flusher = metadata_writer.commit().unwrap();
```

Comment:
our benchmarks probably don't need to flush?

Reply:
I think we do, otherwise a new reader can't query